_hardcoded/tracing.go

```diff
@@ -145,15 +147,16 @@ func MuxServerMiddleware(serviceName string) func(http.Handler) http.Handler {
 			// Add baggage to the context
 			ctx = baggage.ContextWithBaggage(ctx, bags)
 
-			// Log if sampled
-			if s.SpanContext().IsSampled() {
-				logger.FromContext(ctx).AddContext("sampled", "true")
-			} else {
-				logger.FromContext(ctx).AddContext("sampled", "false")
-			}
-
 			// Encode the trace/span ids in the DD format
 			if sc := s.SpanContext(); sc.HasTraceID() {
+
+				// Log if sampled
+				if s.SpanContext().IsSampled() {
+					logger.FromContext(ctx).AddContext("sampled", "true")
+				} else {
+					logger.FromContext(ctx).AddContext("sampled", "false")
+				}
+
```

Comment:
No need to log if it's sampled or not if we aren't actually exporting the "sampled" trace.